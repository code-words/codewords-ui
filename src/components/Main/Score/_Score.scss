.Score {
  margin: 20px;
  padding: 20px;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 150px;
  height: 200px;

  .score-heading {
    font-weight: 400;
    font-size: 25px;
    text-shadow: black 1.5px 1.5px 2px;
  }

  .score-count {
    font-size: 78px;
    margin: 15px;
    text-shadow: black 1.5px 1.5px 2px;
  }

  .player-name {
    align-self: flex-start;
    margin: 4px;
    font-weight: 700;
    text-shadow: black 1.5px 1.5px 2px;
  }

  &.blue-score {
    background: $team-blue-bg;
    border: 4px solid $team-blue-border;
  }

  &.red-score {
    background: $team-red-bg;
    border: 4px solid $team-red-border;
  }

  &.active-team-true {
    border: 4px solid $player-active;
  }
}

.fa-crosshairs {
  color: $player-active;
  margin: 0 5px;
  text-shadow: .5px .5px 1px gray;
}