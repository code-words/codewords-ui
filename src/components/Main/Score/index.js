import React from 'react';

const Score = props => {
	const scoreClass = props.team === 'blue' ? 'blue-score' : 'red-score';
	return (
		<article className={`Score ${scoreClass}`}>
			<h3 className="score-heading">Team Score</h3>
			<p className="score-count">{props.score}</p>
			<p className="player-name">{props.players[0]}</p>
			<p className="player-name">{props.players[1]}</p>
		</article>
	);
};

export default Score;
