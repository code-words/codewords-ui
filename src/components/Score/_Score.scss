.Score {
  margin: 20px;
  padding: 20px;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 150px;
  height: 200px;

  .score-heading {
    font-weight: 400;
    font-size: 25px;
  }

  .score-count {
    font-size: 78px;
    margin: 15px;
  }

  .player-name {
    align-self: flex-start;
    margin: 4px;
  }

  &.blue-score {
    background: #5861B4;
    border: 4px solid rgb(36, 36, 108);
  }

  &.red-score {
    background: #B44646;
    border: 4px solid rgb(99, 36, 36);
  }
}

