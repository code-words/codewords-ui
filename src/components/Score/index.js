import React from 'react';

const Score = props => {
  const scoreClass = props.team === 1 ? "blue-score" : "red-score";
  return (
    <article className={`Score ${scoreClass}`}>
      <h3 className="score-heading">Team {props.team} Score</h3>
      <p className="score-count">{props.score}</p>
      <p className="player-name">{props.players[0]}</p>
      <p className="player-name">{props.players[1]}</p>
    </article>
  );
};

export default Score;