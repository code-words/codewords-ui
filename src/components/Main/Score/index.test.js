import React from 'react';
import Score from '.';
import { shallow } from 'enzyme';

describe('Score', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(<Score players={['Justin', 'Lynne']}/>);
    instance = wrapper.instance();

  })

  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})