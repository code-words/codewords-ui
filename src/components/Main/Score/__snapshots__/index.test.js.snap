// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score should match component snapshot 1`] = `ShallowWrapper {}`;
